File: tools/conf/output/ts/MoveClipConfig.ts
class MoveClipConfig {
    private static _dic:Dictionary<MoveClipConfig> = new Dictionary<MoveClipConfig>();
    private static _arr:Array<MoveClipConfig> = new Array<MoveClipConfig>();

	 public Id :number ;
     public Texture :string ;
     public Config :string ;
     public Name :string ;
     public Play :string ;
     public FrameRate :number ;
     public Times :number ;
     public Location : Array<number>  =new Array<number>();
     public Scale : Array<number>  =new Array<number>();
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<MoveClipConfig>{
        return  MoveClipConfig._dic;
    }
    public static get Arr():Array<MoveClipConfig>{
        return MoveClipConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:MoveClipConfig = <MoveClipConfig>json[i];
            MoveClipConfig._arr.push(config);
            MoveClipConfig._dic.add(config.Id,config);
        }
    }
}

File: server/src/game/conf/MoveClipConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type MoveClipConfigElem struct {
	Id int
	Texture string
	Config string
	Name string
	Play string
	FrameRate float32
	Times int
	Location []int
	Scale []float32
	
}

type MoveClipConfig0 struct {
	Dic map[int]*MoveClipConfigElem
	Arr []*MoveClipConfigElem
}

var (
	MoveClipConfig MoveClipConfig0
)

func loadMoveClipConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	MoveClipConfig.Dic = make(map[int]*MoveClipConfigElem)
	MoveClipConfig.Arr = make([]*MoveClipConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &MoveClipConfig.Arr)
	for _,conf := range MoveClipConfig.Arr{
		MoveClipConfig.Dic[conf.Id] = conf
	}
	
}
